docs/proposals/20220221-runtime-SDK.md

```diff
@@ -0,0 +1,802 @@
+a Runtime Hook, you are required to implement a Runtime Extension handling requests according to the
+OpenAPI specification for the Runtime Hook you are interested in.
+
+Runtime Extensions by design are very powerful and flexible, however given that with great power comes
+great responsibility, a few key consideration should always be kept in mind (more details in the following paragraphs):
+
+- Runtime Extension are components that should be designed, written and deployed with great caution given that they
+  can affect the proper functioning of the Cluster API runtime.
+- Cluster administrators should carefully vet any Runtime Extension registration, thus preventing malicious components
+  from being added to the system.
+
+Please note that following similar practices is already commonly accepted in the Kubernetes ecosystem for
+Kubernetes API server admission webhooks, and Runtime Extensions share the same foundation and most of the same
+considerations/concerns apply.
+
+### Implementing Runtime Extensions
+
+As a developer building systems on top of Cluster API, if you want to interact with the Cluster’s lifecycle via a
+Runtime Extension, you are required to implement an HTTP server handling requests according to the
+OpenAPI specification for the Runtime Hook you are interested in.
+
+Please note that Cluster API only enforces input and output parameters types as defined by a Runtime Hook version,
+and developers are fully responsible of all the other elements of the design of a Runtime Extension implementation,
+including:
+
+- To choose which programming language to use; please note that for sake of this proposal Golang is the language
+  of choice, and we are not planning to test/provide tooling/libraries for other languages. Nevertheless, given that
+  we rely on Open API and plain HTTP(s) calls, other languages should just work but support will be provided at
+  best effort.
+- In case it is required to implement more than one extension, to choose if to have a single HTTP server with multiple
+  handlers or separated HTTP servers;
+- To choose if to have a dedicated HTTP server(s) for Runtime Extensions only or if to use the HTTP server for other
+  purposes as well (e.g. metric endpoint).
+
+In case the Runtime Extension is being developed in Golang, the implementer can benefit from importing
+`sigs.k8s.io/cluster-api` and:
+
+- Use Golang types defined under `/runtime` (the types from which the OpenAPI specification has been generated).
+- Use the RuntimeExtension catalog object to generate the skeleton of the HTTP handler for a given Runtime extension.
+  The generated func will take care of scaffolding tasks like Marshal/Unmarshal; the only missing part will be to
+  implement a strongly typed func that contains the actual extension implementation.
+
+E.g.
+_Note: this is pseudo code, meant to demonstrate that implementing a RuntimeExtension requires only minimal scaffolding;
```

Comment:
nit: Markdown renders those two lines in one

> E.g. Note: this is pseudo code, meant to demonstrate that implementing a RuntimeExtension requires only minimal scaffolding; the exact details will be defined during implementation, possibly taking advantage of Golang generics.

(just in case that's not intended)